### DISCUSSION TITLE
Access detector output as raw tensor

### DISCUSSION THREAD

**Discussion by aubreydacunha** [open-mmlab/mmdetection#9271] [Q&A] [answered]
Hello,

I am working on training a model on my own dataset, but I am getting some odd behaviors.  I can debug them myself, but it would be very helpful to actually see the raw tensor output from the model before it gets processed into bounding boxes.  However, every time I try to find that in the code, I get lost in a tangle of multiple models with multiple base classes with multiple versions of `forward` (technically, one version of `forward` that dispatches to multiple implementations based on some non-intuitive configuration settings).  It's a similar sort of issue to [this](https://github.com/open-mmlab/mmdetection/issues/6188).

I'm assuming there isn't a line anywhere as simple as
```
output = model(input)
```
but where would I look to actually see the input and output tensors?

If the answer is model-dependent, I am specifically fine-tuning a YOLOX model.

**Comment by jbwang1997** [comment, marked as answer]
In mmdet 2.x, it needs to modify the code to achieve the function. You can hack in the model and directly return the results of `self.forward(feats)` in [`bbox_head`](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/dense_heads/dense_test_mixins.py#L36).

If you try the newest mmdet 3.0, you can directly get the raw outputs by using `output = model(input, mode='tensor')`.
